database/controllers/dbm.js

```diff
@@ -242,8 +243,37 @@ module.exports = {
     //Description:  This function returns a JSON object of all stations.
     // If the registered value is true then only the registered stations are returned.
     // Otherwise, it only returns unregistered stations.
-    getStations(registered){
-        return true;
+    getStations(regTypeRequest){
+        if (regTypeRequest === undefined){
+            //registered type not specified; return all stations
+            return machine.findAll({
+                raw: true
+            }).then(stations => {
+                return stations;
+            }).catch(err => {
+                return false;
+            });
+        } else if (regTypeRequest === false) {
```

Comment:
This works, but wouldn't you be able to just have a variable to store the `registered` value?
Perhaps something like this to avoid code duplication:
```
var registered_flag = false;
if (regTypeRequest === true)
   registered_flag = true;
return machine.findAll({ where: {registered: registered_flag} }) ............ etc;
```

Reply:
That's a really good point. Revision coming shortly.